PABC.Server.Test/Features/GetApplicationRolesPerEntityType/GetApplicationRolesPerEntityTypeControllerTests.cs

```diff
@@ -1,16 +1,131 @@
+        var domain2 = new Domain { Id = Guid.NewGuid(), EntityTypes = [entityType], Description = string.Empty, Name = "Domain 2" };
+
+        var mappings = new List<Mapping>
         {
-            var controller = new GetApplicationRolesPerEntityTypeController(postgresFixture.DbContext);
-            var result = await controller.Post(new GetApplicationRolesRequest { FunctionalRoleNames = [] });
-            Assert.NotNull(result);
-        }
+            new() { Id = Guid.NewGuid(), FunctionalRole = functionalRole, ApplicationRole = applicationRole, Domain = domain1 },
+            new() { Id = Guid.NewGuid(), FunctionalRole = functionalRole, ApplicationRole = applicationRole, Domain = domain2 } // will result in duplicate result in the query if we don't handle it explicitly
+        };
+
+        _dbContext.AddRange(functionalRole, applicationRole, entityType, domain1, domain2);
+        _dbContext.AddRange(mappings);
+        await _dbContext.SaveChangesAsync();
+    }
+
+    [Fact]
+    public async Task Post_ReturnsExpectedData_ForValidFunctionalRole()
+    {
+        var result = await CreateController().Post(CreateRequest(ValidFunctionalRole));
+        var response = Assert.IsType<GetApplicationRolesResponse>(result.Value);
+        var role = Assert.Single(response.Results).ApplicationRoles.Single();
+
+        Assert.Equal(ApplicationRoleName, role.Name);
+        Assert.Equal(ApplicationName, role.Application);
+    }
+
+    [Fact]
+    public async Task Post_ReturnsEmpty_WhenFunctionalRoleDoesNotExist()
+    {
+        var result = await CreateController().Post(CreateRequest(InvalidFunctionalRole));
+        var response = Assert.IsType<GetApplicationRolesResponse>(result.Value);
+        Assert.Empty(response.Results);
+    }
+
+    [Fact]
+    public async Task Post_ReturnsEmpty_WhenFunctionalRoleNamesIsEmpty()
+    {
+        var result = await CreateController().Post(CreateRequest());
+        var response = Assert.IsType<GetApplicationRolesResponse>(result.Value);
+        Assert.Empty(response.Results);
+    }
+
+    [Fact]
+    public async Task Post_HandlesDuplicateFunctionalRoleNames_WithoutDuplication()
+    {
+        var result = await CreateController().Post(CreateRequest(ValidFunctionalRole, ValidFunctionalRole));
+        var response = Assert.IsType<GetApplicationRolesResponse>(result.Value);
+
+        Assert.Single(response.Results);
+        Assert.Single(response.Results.First().ApplicationRoles);
+    }
+
+    [Fact]
+    public async Task Post_HandlesDuplicateMappings_WithoutDuplicateApplicationRoles()
+    {
+        var result = await CreateController().Post(CreateRequest(ValidFunctionalRole));
+        var response = Assert.IsType<GetApplicationRolesResponse>(result.Value);
+
+        Assert.Single(response.Results);
+        Assert.Single(response.Results.First().ApplicationRoles);

```

Comment:
I like the pattern you used above, can you use it in all the test methods where it's applicable? `Assert.Single()` will give us a better error message than `.First()` or `.Single()` on the actual collection
```suggestion
        var singleResult = Assert.Single(response.Results);
        Assert.Single(singleResult.ApplicationRoles);

```

Reply:
Will do

Reply:
I have addressed the comment